src/UNL/VisitorChat/Util.php

```diff
@@ -0,0 +1,36 @@
+<?php
+namespace UNL\VisitorChat;
+
+class Util
+{
+    public static function isAllowedDomain($url, array $domains)
+    {
+        if (!$parts = parse_url($url)) {
+            //it isn't a valid URL, don't allow
+            return false;
+        }
+        
+        if (!isset($parts['host'])) {
+            //it isn't a valid host, don't allow
+            return false;
+        }
+
+        $possibilities = array();
+        foreach ($domains as $domain) {
+            //Root domains are okay
+            $possibilities[] = '^' . preg_quote($domain, "/");
+            //Sub-domains are okay
+            $possibilities[] = '.*\.' . preg_quote($domain, "/");
+        }
+        
+        $regex = implode($possibilities, "|");
+        
+        $regex = "/(" . $regex . ")$/i";
```

Comment:
I was just thinking that `$regex = "/^(.*\.)?(" . $regex . ")$/i"` might be easier than having to have to add a possibility for each domain.